File: motomanager/backend/MotoManager.Domain/Entities/ServiceOrderMaterial.cs
namespace MotoManager.Domain.Entities;

public class ServiceOrderMaterial
{
    public long Id { get; set; }
    public long ServiceOrderId { get; set; }
    public long MaterialId { get; set; }
    public decimal Quantity { get; set; }
    public decimal PricePerUnit { get; set; }
    public decimal TotalPrice { get; set; }

    public ServiceOrder? ServiceOrder { get; set; }
    public Material? Material { get; set; }
}

File: motomanager/frontend/motomanager-web/src/app/models/service-order-material.ts
export interface ServiceOrderMaterial {
  id: number;
  serviceOrderId: number;
  materialId: number;
  materialName: string;
  unitOfMeasureName?: string;
  quantity: number;
  pricePerUnit: number;
  totalPrice: number;
}

export interface AddServiceOrderMaterialRequest {
  materialId: number;
  quantity: number;
  pricePerUnit: number;
}

export interface UpdateServiceOrderMaterialRequest {
  quantity: number;
  pricePerUnit: number;
}
